provider/datagen/src/provider.rs

```diff
@@ -68,9 +68,13 @@ impl DatagenProvider {
             once_cell::sync::OnceCell::new();
         SINGLETON
             .get_or_init(|| {
+                let maybe_var = std::env::var("CARGO_MANIFEST_DIR").ok();
```

Comment:
I really don't like the complexity of this. 6 lines to define a path is 5 too many.